<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PageSampledData</class>
 <widget class="QWidget" name="PageSampledData">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>613</width>
    <height>472</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="tabShape">
      <enum>QTabWidget::Triangular</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Current</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QStackedWidget" name="currentStack">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="page">
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QCustomPlot" name="currentPlot" native="true">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_14">
               <item>
                <widget class="QCheckBox" name="showCurrent1Box">
                 <property name="text">
                  <string>PH 1</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="showCurrent2Box">
                 <property name="text">
                  <string>PH 2</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="showCurrent3Box">
                 <property name="text">
                  <string>PH 3</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="showMcTotalCurrentBox">
                 <property name="text">
                  <string>MC total</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="showPosCurrentBox">
                 <property name="text">
                  <string>Position</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="showPhaseBox">
                 <property name="toolTip">
                  <string>Show FOC motor phase</string>
                 </property>
                 <property name="text">
                  <string>Phase</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer_3">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>0</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_2">
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QSplitter" name="splitter">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <widget class="QWidget" name="layoutWidget_3">
              <layout class="QVBoxLayout" name="verticalLayout_8">
               <item>
                <widget class="QCustomPlot" name="filterResponsePlot" native="true"/>
               </item>
               <item>
                <widget class="QCheckBox" name="filterLogScaleBox">
                 <property name="text">
                  <string>Logscale</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="layoutWidget_4">
              <layout class="QVBoxLayout" name="verticalLayout_10">
               <item>
                <widget class="QCustomPlot" name="filterPlot" native="true"/>
               </item>
               <item>
                <widget class="QCheckBox" name="filterScatterBox">
                 <property name="text">
                  <string>Scatterplot</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QComboBox" name="plotModeBox">
           <property name="toolTip">
            <string>Select what to plot</string>
           </property>
           <item>
            <property name="text">
             <string>Time Plot</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>FFT Plot</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Filter Plot</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Filter Data</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <property name="leftMargin">
             <number>1</number>
            </property>
            <property name="topMargin">
             <number>6</number>
            </property>
            <property name="rightMargin">
             <number>1</number>
            </property>
            <property name="bottomMargin">
             <number>6</number>
            </property>
            <item>
             <widget class="QComboBox" name="filterBox">
              <item>
               <property name="text">
                <string>No Filter</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>FIR Filter</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Mean Filter</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="currentFilterFreqBox">
              <property name="toolTip">
               <string>FIR filter stop frequency</string>
              </property>
              <property name="prefix">
               <string>F_St: </string>
              </property>
              <property name="decimals">
               <number>3</number>
              </property>
              <property name="maximum">
               <double>0.500000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.005000000000000</double>
              </property>
              <property name="value">
               <double>0.050000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="currentFilterTapBox">
              <property name="toolTip">
               <string>FIR filter taps</string>
              </property>
              <property name="prefix">
               <string>Taps: </string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
              <property name="value">
               <number>6</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="compDelayBox">
              <property name="toolTip">
               <string>Compensate for filter delay</string>
              </property>
              <property name="text">
               <string>Delay Comp</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="hammingBox">
              <property name="toolTip">
               <string>Use Hamming window</string>
              </property>
              <property name="text">
               <string>Hamming</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="fftFreqBox">
           <property name="toolTip">
            <string>Sampling frequency to calculate FFT time axis at</string>
           </property>
           <property name="prefix">
            <string>Fs: </string>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>200000</number>
           </property>
           <property name="singleStep">
            <number>1000</number>
           </property>
           <property name="value">
            <number>40000</number>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>BEMF</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QCustomPlot" name="voltagePlot" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QCheckBox" name="showPh1Box">
           <property name="text">
            <string>PH 1</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="showPh2Box">
           <property name="text">
            <string>PH 2</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="showPh3Box">
           <property name="text">
            <string>PH 3</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="showVirtualGndBox">
           <property name="text">
            <string>Virtual Ground</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="showPosVoltageBox">
           <property name="text">
            <string>Position</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="showPhaseVoltageBox">
           <property name="toolTip">
            <string>Show FOC motor phase</string>
           </property>
           <property name="text">
            <string>Phase</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QCheckBox" name="truncateBox">
           <property name="toolTip">
            <string>Truncate conducting phases</string>
           </property>
           <property name="text">
            <string>Truncate</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="spacing">
      <number>3</number>
     </property>
     <item>
      <widget class="QPushButton" name="sampleNowButton">
       <property name="toolTip">
        <string>Sample now</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sampleStartButton">
       <property name="toolTip">
        <string>Sample when the motor starts</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="iconSize">
        <size>
         <width>16</width>
         <height>16</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sampleTriggerStartButton">
       <property name="toolTip">
        <string>Triggered sampling when the motor starts</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="iconSize">
        <size>
         <width>30</width>
         <height>16</height>
        </size>
       </property>
       <property name="shortcut">
        <string>P</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sampleTriggerFaultButton">
       <property name="toolTip">
        <string>Triggered sampling at fault codes</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="iconSize">
        <size>
         <width>30</width>
         <height>16</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sampleTriggerStartNosendButton">
       <property name="toolTip">
        <string>Triggered sampling when the motor starts without sending the samples automatically</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="iconSize">
        <size>
         <width>40</width>
         <height>16</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sampleTriggerFaultNosendButton">
       <property name="toolTip">
        <string>Triggered sampling at fault codes without sending the samples automatically</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="iconSize">
        <size>
         <width>40</width>
         <height>16</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sampleLastButton">
       <property name="toolTip">
        <string>Read last samples</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="sampleStopButton">
       <property name="toolTip">
        <string>Stop sampling</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="rawBox">
       <property name="toolTip">
        <string>Use raw ADC values instead of scaled voltages and currents.</string>
       </property>
       <property name="text">
        <string>Raw</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="loadDataButton">
       <property name="toolTip">
        <string>Load samples from CSV-file</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="saveDataButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Save samples to CSV file...</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>3</number>
     </property>
     <item>
      <widget class="QSpinBox" name="samplesBox">
       <property name="toolTip">
        <string>Number of samples total, or number of samples before trigger in triggered modes</string>
       </property>
       <property name="prefix">
        <string>Samp: </string>
       </property>
       <property name="maximum">
        <number>1600</number>
       </property>
       <property name="singleStep">
        <number>100</number>
       </property>
       <property name="value">
        <number>800</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="decimationBox">
       <property name="toolTip">
        <string>Decimation (skip samples to fit more data)</string>
       </property>
       <property name="prefix">
        <string>Dec: </string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QProgressBar" name="sampProgBar">
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="zoomHButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <pointsize>3</pointsize>
        </font>
       </property>
       <property name="toolTip">
        <string>Enable horizontal zoom</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="iconSize">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="zoomVButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <pointsize>3</pointsize>
        </font>
       </property>
       <property name="toolTip">
        <string>Enable vertical zoom</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="iconSize">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="rescaleButton">
       <property name="toolTip">
        <string>Rescale plots to fit</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="iconSize">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>widgets/qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
